lib/puppet_x/puppetlabs/scheduled_task/trigger.rb

```diff
@@ -106,6 +106,207 @@ class V1
+      end
+
+      def self.bitmask_to_indexes(bitmask)
+        bitmask = Integer(bitmask)
+        if (bitmask < 0 || bitmask > 0xFFFFFFFF)
+          raise ArgumentError.new("bitmask must be specified as an integer from 0 to #{0xFFFFFFFF.to_s(10)}")
+        end
+
+        (0..31).select do |bit_index|
+          bit_to_check = 1 << bit_index
+          # given position is set in the bitmask
+          (bitmask & bit_to_check) == bit_to_check
+        end.map do |bit_index|
+          # Day 32 has the special meaning of "the last day of the
+          # month", no matter how many days there are in the month.
+          bit_index == 31 ? 'last' : bit_index + 1
+        end
+      end
+    end
+
+    class Month
+      # https://msdn.microsoft.com/en-us/library/windows/desktop/aa381918(v=vs.85).aspx
+      TASK_JANUARY      = 0x1
+      TASK_FEBRUARY     = 0x2
+      TASK_MARCH        = 0x4
+      TASK_APRIL        = 0x8
+      TASK_MAY          = 0x10
+      TASK_JUNE         = 0x20
+      TASK_JULY         = 0x40
+      TASK_AUGUST       = 0x80
+      TASK_SEPTEMBER    = 0x100
+      TASK_OCTOBER      = 0x200
+      TASK_NOVEMBER     = 0x400
+      TASK_DECEMBER     = 0x800
+
+      MONTHNUM_CONST_MAP = {
+        1  => TASK_JANUARY,
+        2  => TASK_FEBRUARY,
+        3  => TASK_MARCH,
+        4  => TASK_APRIL,
+        5  => TASK_MAY,
+        6  => TASK_JUNE,
+        7  => TASK_JULY,
+        8  => TASK_AUGUST,
+        9  => TASK_SEPTEMBER,
+        10 => TASK_OCTOBER,
+        11 => TASK_NOVEMBER,
+        12 => TASK_DECEMBER,
+      }.freeze
+
+      def self.indexes_to_bitmask(month_indexes)
+        month_indexes = [month_indexes].flatten.map { |m| Integer(m) rescue m }
+        invalid_months = month_indexes - MONTHNUM_CONST_MAP.keys
+        raise ArgumentError.new('Month must be specified as an integer in the range 1-12') unless invalid_months.empty?
+
+        month_indexes.inject(0) { |bitmask, month_index| bitmask |= MONTHNUM_CONST_MAP[month_index] }
+      end
+
+      def self.bitmask_to_indexes(bitmask)
+        bitmask = Integer(bitmask)
+        if (bitmask < 0 || bitmask > 0xFFF)
+          raise ArgumentError.new("bitmask must be specified as an integer from 0 to #{0xFFF.to_s(10)}")
+        end
+
+        MONTHNUM_CONST_MAP.values.each_with_object([]) do |day, indexes|
+          indexes << MONTHNUM_CONST_MAP.key(day) if bitmask & day != 0
+        end
+      end
+    end
+
+    class Occurrence
+      # https://msdn.microsoft.com/en-us/library/windows/desktop/aa381950(v=vs.85).aspx
+      TASK_FIRST_WEEK   = 1
+      TASK_SECOND_WEEK  = 2
+      TASK_THIRD_WEEK   = 3
+      TASK_FOURTH_WEEK  = 4
+      TASK_LAST_WEEK    = 5
+
+      CONST_MAP = {
+        'first'  => TASK_FIRST_WEEK,
+        'second' => TASK_SECOND_WEEK,
+        'third'  => TASK_THIRD_WEEK,
+        'fourth' => TASK_FOURTH_WEEK,
+        'last'   => TASK_LAST_WEEK,
+      }
```

Comment:
freeze me ... also, better name than `CONST_MAP`?

Reply:
I was ok with const map as it's internal to the class so easy to reason about.